lib/api.js

```diff
@@ -211,10 +216,15 @@ class Api extends Emittery {
 
 				const options = {
 					...apiOptions,
+					filter,
 					providerStates,
 					recordNewSnapshots: !isCi,
 					// If we're looking for matches, run every single test process in exclusive-only mode
-					runOnlyExclusive: apiOptions.match.length > 0 || runtimeOptions.runOnlyExclusive === true
+					runOnlyExclusive: apiOptions.match.length > 0 || runtimeOptions.runOnlyExclusive === true,
+					runOnlySelectedLines: filter.some(({pattern, lineNumbers}) =>
+						micromatch.isMatch(globs.normalizeFileForMatching(apiOptions.projectDir, file), pattern) &&
+						lineNumbers !== null
```

Comment:
Let's check `lineNumbers` first so we don't match unnecessarily.

```suggestion
						lineNumbers !== null && micromatch.isMatch(globs.normalizeFileForMatching(apiOptions.projectDir, file), pattern)
```